client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/CRUDDocumentationIT.java

```diff
@@ -318,4 +325,87 @@ public void onFailure(Exception e) {
             // end::delete-conflict
         }
     }
+
+    public void testBulk() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+        {
+            BulkRequest bulkRequest = new BulkRequest()
+                    .add(new IndexRequest("index", "type", "1").source(Requests.INDEX_CONTENT_TYPE, "field", "value"))
+                    .add(new IndexRequest("index", "type", "2").source(Requests.INDEX_CONTENT_TYPE,"field", "value"));
+            BulkResponse bulkResponse = client.bulk(bulkRequest);
+            assertFalse(bulkResponse.hasFailures());
+        }
+        {
+            // tag::bulk-request
+            BulkRequest request = new BulkRequest(); // <1>
+            request.add(new IndexRequest("index", "type", "0").source(XContentType.JSON,"field", "value")); // <2>
+            request.add(new UpdateRequest("index", "type", "1").doc(XContentType.JSON,"field", "value")); // <3>
+            request.add(new DeleteRequest("index", "type", "2")); // <4>
+            // end::bulk-request
+
+            // tag::bulk-execute
+            BulkResponse bulkResponse = client.bulk(request);
+            // end::bulk-execute
+            assertSame(bulkResponse.status(), RestStatus.OK);
+            assertFalse(bulkResponse.hasFailures());
+
+            // tag::bulk-response
+            for (BulkItemResponse bulkItemResponse : bulkResponse) { // <1>
+                DocWriteResponse itemResponse = bulkItemResponse.getResponse(); // <2>
+
+                if (itemResponse instanceof IndexResponse) { // <3>
```

Comment:
maybe we could do the if based on the opType? we'd still need the instanceOf though...